src/datatablerow/datatablerow.jl

```diff
@@ -65,3 +72,47 @@ function Base.isequal(r1::DataTableRow, r2::DataTableRow)
     end
     return true
 end
+
+# internal method for comparing the elements of the same data table column
+isequal_colel(col::AbstractArray, r1::Int, r2::Int) =
+    (r1 == r2) || isequal(Base.unsafe_getindex(col, r1), Base.unsafe_getindex(col, r2))
+
+function isequal_colel{T}(col::Union{NullableArray{T}, AbstractNullableCategoricalArray{T}}, r1::Int, r2::Int)
+    (r1 == r2) && return true
+    isequal(col[r1], col[r2])
+end
+
+isequal_colel(a::Any, b::Any) = isequal(a, b)
+isequal_colel(a::Nullable, b::Any) = !isnull(a) & isequal(unsafe_get(a), b)
+isequal_colel(a::Any, b::Nullable) = isequal_colel(b, a)
+isequal_colel(a::Nullable, b::Nullable) = isequal(a, b)
+
+function isequal_row(dt1::AbstractDataTable, r1::Int, dt2::AbstractDataTable, r2::Int)
+    (dt1 === dt2) && return isequal(dt1[r1], dt1[r2])
```

Comment:
I think you're right: `isequal_row()` should handle all the cases and also be called by `isequal(dr1, dr2)`. I wanted to review the `DataFramesRow` API for my PR before giving another try of merging it to the master, but @cjprybol was faster :)